product-scenarios/1-manage-users-roles/1.1-user-registration/1.1.2-user-registration-by-admin/1.1.2.1-user-registration-through-apis/1.1.2.1.2-scim-2.0/src/test/java/org/wso2/identity/scenarios/test/scim2/ProvisionEmailUserSCIM2TestCase.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.scenarios.test.scim2;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.identity.scenarios.commons.ScenarioTestBase;
+
+import org.wso2.identity.scenarios.commons.util.Constants;
+import org.wso2.identity.scenarios.commons.util.SCIMProvisioningUtil;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+import static org.wso2.identity.scenarios.commons.util.IdentityScenarioUtil.getJSONFromResponse;
+
+
+
+public class ProvisionEmailUserSCIM2TestCase extends ScenarioTestBase {
+
+    private CloseableHttpClient client;
+    private String userNameResponse;
+    private final String EMAIL_ID="scim2user@wso2.com";
+    private String userId;
+
+    HttpResponse response;
+
+
+    @BeforeClass(alwaysRun = true)
```

Comment:
Why dont we have @AfterClass method for cleanup of ressources?

Reply:
the cleanup method was marked as dependsOnMethods from the main test method so that if that method fails the cleanup wont run can we use dependsOnMethods with @afterclass

Reply:
changed to use @afterclass, thanks